### PR TITLE
Set Irrlichts loglevel to Minetests loglevel

### PR DESCRIPTION
**Pull request by adrido**
with debug_log_level set to warning:

before:
```
2017-04-02 10:13:35: WARNING[Main]: Couldn't find a locale directory!
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/textures/base/pack/menu_header.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/games/king_arthurs_game/menu/overlay.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/games/king_arthurs_game/menu/header.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/games/king_arthurs_game/menu/icon.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/games/minetest_game/menu/icon.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/games/minimal/menu/icon.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/games/minetest_game/menu/header.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/games/king_arthurs_game/menu/overlay.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/games/king_arthurs_game/menu/header.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/games/minimal/menu/background.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/games/minetest_game/menu/header.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/games/king_arthurs_game/menu/overlay.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/games/king_arthurs_game/menu/header.png
Ignoring resize operation to (0 0)
Resizing window (1000 750)
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/textures/base/pack/menu_header.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/textures/base/pack/blank.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/textures/base/pack/server_flags_favorite.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/textures/base/pack/server_ping_4.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/textures/base/pack/server_ping_3.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/textures/base/pack/server_ping_2.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/textures/base/pack/server_ping_1.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/textures/base/pack/server_flags_creative.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/textures/base/pack/server_flags_damage.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/textures/base/pack/server_flags_pvp.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/games/king_arthurs_game/menu/overlay.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/games/king_arthurs_game/menu/header.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/textures/base/pack/menu_header.png
Loaded texture: D:/Users/Doll/Documents/Projects/Minetest/textures/base/pack/no_screenshot.png
```
after:
` 2017-04-02 10:26:14: WARNING[Main]: Couldn't find a locale directory!`

this would solve: https://github.com/minetest/minetest/issues/4255
a little bit related: https://github.com/minetest/minetest/pull/5416

I created a separate PR, because this is independent of https://github.com/minetest/minetest/pull/5416 and may be merged faster.

### CODE REVIEW

**Comment by red-001** [comment]
Looks like this broke the server-only build, I assume it's something to do with it not including irrlicht

**Comment by adrido** [comment]
Thanks :-) Is there a way to re-run the check?

**Comment by SmallJoker** [comment]
@adrido Amend nothing and force push ;)

File: src/client/clientlauncher.cpp [hunk 1]
```diff
@@ -96,6 +96,31 @@ bool ClientLauncher::run(GameParams &game_params, const Settings &cmd_args)
 		return false;
 	}
 
+#ifndef SERVER
+	// Set Irrlicht LogLevel
+	switch (Logger::stringToLevel(g_settings->get("debug_log_level"))) {
+	case LL_NONE:
+		device->getLogger()->setLogLevel(ELL_NONE);
```
Comment (SmallJoker) [inline src/client/clientlauncher.cpp]: Use `irr::` namespace

File: src/client/clientlauncher.h [hunk 2]
```diff
@@ -23,6 +23,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "irrlichttypes_extrabloated.h"
 #include "client/inputhandler.h"
 #include "gameparams.h"
+#include "ILogger.h"
```
Comment (SmallJoker) [inline src/client/clientlauncher.h]: Not used in header, should be moved to the cpp file.

**Comment by adrido** [comment]
Thanks for your suggestions, but it doesnt help with that 3 compilers.
It seems it doesnt know the ‘ELL_DEBUG’ or `irr::ELL_DEBUG` but it should be there: http://irrlicht.sourceforge.net/docu/namespaceirr.html#aa2d1cac68606a25ed24cfffccfa30a92
Might there something wrong with the build environment? It seems to compile fine on 5 of 8 environments.

**Comment by adrido** [comment]
It compiles now. the problem was that ELL_DEBUG was added in Irrrlicht 1.8 that was not installed on that environments. 
Its now ready for testing, reviewing and (hopefully) merging :-)

File: src/client/clientlauncher.cpp [hunk 3]
```diff
@@ -36,6 +40,11 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "clientlauncher.h"
 #include "version.h"
 
+// irr::ELL_DEBUG was added in irrlicht 1.8
+#if (IRRLICHT_VERSION_MAJOR == 1 && IRRLICHT_VERSION_MINOR < 8)
+#define ELL_DEBUG ELL_INFORMATION
```
Comment (sfan5) [inline src/client/clientlauncher.cpp]: this is a horrible way of doing this, move the `#ifdef` down into the actual code

**Comment by rubenwardy** [comment]
I suggest adding irrlicht_debug_log_level. If it's unspecified, then it defaults to debug_log_level

**Comment by adrido** [comment]
@rubenwardy keep it simple 😉 . IMO there is no benefit of adding another setting. It may only cause confusion to the player because they may not even know which stuff is added by the Irrlicht engine or by Minetest. 
Thanks for your suggestion, but I ill not add it for above reason.

**Comment by rubenwardy** [comment]
How much more output does lower debug levels make irrlicht spew? If it's neglible then I'm fine with it

**Comment by sofar** [comment]
review standoff? this either needs changing or the reviewers need to dismiss their reviews. It's unclear to me what is needed to make this go forward.

**Comment by sfan5** [comment]
I think @rubenwardy's idea with a seperate `irrlicht_log_level` is a good compromise on this.

**Comment by SmallJoker** [comment]
Sure, why not? Let's add another setting. (semi-sarcasm)

**Comment by nerzhul** [comment]
can you split params @adrido ?

**Comment by adrido** [comment]
😕 What params?

**Comment by nerzhul** [comment]
change debug_log_level to use a new setting: irrlicht_log_level

**Comment by adrido** [comment]
Well, I'm not a friend of another setting. But if all devs want a additional setting for this, i can add it. But not this week and probably not next week, my time-plan is full.

**Comment by ShadowNinja** [comment]
Thumbs down from me, because this is just a workaround for an Irrlicht bug (#3255).
Minetest leaves Irrlicht at its max log level because all Irrlicht messages are filtered through the logging system (except when they're not, due said Irrlicht bug...)

This means that the console and `debug.txt` can have different log levels.  If you apply this patch it will filter messages to all logging destinations regardless of their logging settings (in this case it will apply the `debug.txt` level to both the console and `debug.txt`).

**Comment by adrido** [comment]
Great :-) less work for me.

**Pull request closed without merging**